### PR TITLE
Remove x/cherami.

### PR DESCRIPTION
**Pull request by breerly**
Instead of graduating the Cherami transport out of x/, I want to remove it.

* We've decided to re-evaluate the Oneway launch in 2018 so that we can get Unary and Streaming right.
* We haven't trialed with any customers yet.
* We can always resurrect this transport.

I don't want to keep aspirational features hanging around: x/ is for trialing features with customers.

This should speed up the test suite, improve coverage, and reduce the overall cognitive load of developing YARPC.

### CODE REVIEW

**Comment by peter-edge** [comment]
If you wanna really go for broke, remove redis too, this will allow big
time speed up

**Comment by breerly** [comment]
@peter-edge will remove x/redis in another PR

**Comment by peter-edge** [comment]
Feel free to assign me a task to rework code coverage in Travis to not use
docker compose, if this doesn't include that change

File: yarpcconfig/doc.go [hunk 1]
```diff
@@ -144,16 +141,16 @@
...
 // 	  unary:
-// 	    http:
-// 	      url: http://127.0.0.1:8082/yarpc
+// 	    tchannel:
+//        peer: 127.0.0.1:4040
 // 	  oneway:
-// 	    cherami:
-// 	      destination: /keyvalue/requests
+// 	    http:
+//        url: http://127.0.0.1:8080/
```
Comment (abhinav) [inline yarpcconfig/doc.go]: incorrect indentation for `peer` and `url`.

**Comment by breerly** [comment]
I'll followup with another PR that removes the tabs in the docs across this file

**Pull request merged**